bionic/deriver.py

```diff
@@ -601,8 +591,74 @@ def __init__(self, task):
 
         self.is_complete = False
 
+    def incomplete_dep_states(self):
+        return [dep_state for dep_state in self.dep_states if not dep_state.is_complete]
+
+    @property
     def is_blocked(self):
-        return not all(parent.is_complete for parent in self.parents)
+        return len(self.incomplete_dep_states()) > 0
 
     def __repr__(self):
         return f"TaskState({self.task!r})"
+
+
+class Blockage:
```

Comment:
We can maybe say what this blocks, how about `TaskBlockage` / `TaskStateBlockage`?